src/raiden_libs/service_registry.py

```diff
@@ -180,5 +187,55 @@ def update_service_url(
         )
 
 
+@blockchain_options(contracts=[CONTRACT_DEPOSIT])
+@cli.command("withdraw")
+@click.option(
+    "--to", type=str, callback=validate_address, help="Target address for withdrawn tokens"
+)
+@common_options("service_registry")
+def withdraw(
+    private_key: str,
+    state_db: str,  # pylint: disable=unused-argument
+    web3: Web3,
+    contracts: Dict[str, Contract],
+    start_block: BlockNumber,  # pylint: disable=unused-argument
+    to: str,
+) -> None:
+    """
+    Withdraw tokens deposited to the ServiceRegistry.
+    """
+    # Add middleware to sign transactions by default
+    web3.middleware_onion.add(construct_sign_and_send_raw_middleware(private_key))
+
+    log.info("Using RPC endpoint", rpc_url=get_web3_provider_info(web3))
+    deposit_contract = contracts[CONTRACT_DEPOSIT]
+
+    # Check usage of correct key
+    withdrawer = deposit_contract.functions.withdrawer().call()
+    caller_address = private_key_to_address(private_key)
+    if to_canonical_address(withdrawer) != caller_address:
+        log.error(
+            "You must used the key used to deposit when withdrawing",
+            expected=withdrawer,
+            actual=caller_address,
```

Comment:
This should be checksum encoded

Reply:
Right, I lost that when introducing the `caller_address` variable.